arches_component_lab/src/arches_component_lab/widgets/DatePickerWidget/DatePickerWidget.vue

```diff
@@ -0,0 +1,40 @@
+<script setup lang="ts">
+import DatePickerWidgetEditor from "@/arches_component_lab/widgets/DatePickerWidget/components/DatePickerWidgetEditor.vue";
+import DatePickerWidgetViewer from "@/arches_component_lab/widgets/DatePickerWidget/components/DatePickerWidgetViewer.vue";
+
+import { EDIT, VIEW } from "@/arches_component_lab/widgets/constants.ts";
+
+import type { WidgetMode } from "@/arches_component_lab/widgets/types.ts";
+import type { DateDatatypeCardXNodeXWidgetData } from "@/arches_component_lab/datatypes/date/types.ts";
+import type { NodeData } from "@/arches_component_lab/types.ts";
+
+defineProps<{
+    mode: WidgetMode;
+    nodeAlias: string;
+    graphSlug: string;
+    cardXNodeXWidgetData: DateDatatypeCardXNodeXWidgetData;
+    value: NodeData | null | undefined;
+}>();
+
+const emit = defineEmits(["update:isDirty", "update:value"]);
+</script>
+
+<template>
+    <DatePickerWidgetEditor
+        v-if="mode === EDIT"
+        :card-x-node-x-widget-data="
+            cardXNodeXWidgetData as DateDatatypeCardXNodeXWidgetData
+        "
+        :node-alias="nodeAlias"
+        :value="value"
+        @update:value="emit('update:value', $event)"
+        @update:is-dirty="emit('update:isDirty', $event)"
```

Comment:
we _could_ use magic `v-bind="$attrs"` here to not need the widget designer to define emits in their components, but I'm unsure if that's a good idea. We're doing it in the editor for all widgets to minimize friction, but I'm not married to the idea and could reverse if we wanna be more explicit

Reply:
I'm of 2 minds on this.  I like the explicitness, but if these emits are really just boilerplate and universally needed (which they appear to be) then I might lean towared using `v-bind:"$attr"`